### ISSUE TITLE
Media: Replace spaces and special characters in file uploads

### ISSUE THREAD

**Issue by mshaver** [thinkshout/bene#233]
## Overview
Many Bene clients upload files to reference in content on their site. Since many do this without cleaning up file names, they end up with file paths that are less than ideal. We should put in place a way of cleaning these file names up upon upload. Replacing spaces and special characters are the most important. 

## Dev [5 hrs]

- [ ] First, try to see if an existing module meets the needs of this issue. I feel like https://www.drupal.org/project/media_entity_download would be a good solution for Documents.
- [ ] If not existing module works, extend the core File/Image widget to add a input element that allows for a filename to be changed on upload.
- [ ] After a file is uploaded, use the content of this input element to move the file.

## This ticket is done when:
- [ ] Content editors can configure a media entity's filename, either with custom code or a contrib module.

**Comment by mortenson** [comment]
@mshaver You mention `Replacing spaces and special characters` - does this imply that we can modify filenames programmatically on upload without any manual intervention? If users need to manually change a filename that would be a different kind of task. Thanks!

**Comment by mshaver** [comment]
@mortenson, yes just on file upload. Essentially making sure once files are uploaded they have filenames that won't break when downloading on some computers. I'm pretty sure there's a module for this! ;)
